pkg/cmd/operator/webhooks_test.go

```diff
@@ -0,0 +1,237 @@
+		t.Fatalf("can't create a file: %v", err)
+	}
+
+	wo := WebhookOptions{
+		TLSCertFile:                    TLSCertFile,
+		TLSKeyFile:                     TLSKeyFile,
+		Port:                           5000,
+		InsecureGenerateLocalhostCerts: false,
+	}
+	err = wo.Complete()
+	if err != nil {
+		t.Fatalf("can't complete WebhookOptions: %v", err)
+	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		err := wo.run(ctx, genericclioptions.IOStreams{})
+		if err != nil {
+			t.Errorf("can't run webhook server: %v", err)
+		}
+	}()
+
+	encodedCa, err := cc.getEncodedCA()
+	if err != nil {
+		t.Fatalf("can't get encoded CA: %v", err)
+	}
+
+	certpool := x509.NewCertPool()
+	certpool.AppendCertsFromPEM(encodedCa)
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: certpool,
+			},
+		},
+	}
+
+	addr := fmt.Sprintf("https://localhost:%d/readyz", wo.Port)
+
+	err = wait.PollImmediate(1*time.Second, 15*time.Second, func() (bool, error) {
+		resp, err := client.Get(addr)
+		return err == nil && reflect.DeepEqual(resp.TLS.PeerCertificates[0].SerialNumber, certSerialNumber), nil
+	})
+	if err != nil {
+		t.Fatalf("can't poll immediatelly: %v", err)
+	}
+
+	replaceCertSerialNumber := big.NewInt(2)
+	encodedReplaceCert, err := cc.newEncodedCertificate(replaceCertSerialNumber)
+	if err != nil {
+		t.Fatalf("can't create new encoded replace certificate: %v", err)
+	}
+	if err = createFileWithContent(wo.TLSCertFile, encodedReplaceCert.cert); err != nil {
+		t.Fatalf("can't create a file: %v", err)
+	}
+	if err = createFileWithContent(wo.TLSKeyFile, encodedReplaceCert.privateKey); err != nil {
+		t.Fatalf("can't create a file: %v", err)
+	}
+
+	err = wait.PollImmediate(1*time.Second, 3*time.Second, func() (bool, error) {
+		resp, err := client.Get(addr)
+		return err == nil && reflect.DeepEqual(resp.TLS.PeerCertificates[0].SerialNumber, replaceCertSerialNumber), nil
+	})
+	if err != nil {
+		t.Errorf("certificate wasn't updated: %v", err)
+	}
+}
+
+type CertificateCreator struct {
+	ca           *x509.Certificate
+	caPrivateKey *rsa.PrivateKey
+}
+
+type encodedCertificate struct {
+	cert       []byte
+	privateKey []byte
+}
+
+func newCertificateCreator() (*CertificateCreator, error) {
+	now := time.Now()
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(0),
+		NotBefore:    now,
+		NotAfter:     now.Add(1 * time.Hour),
+
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+		IsCA:                  true,
+
+		DNSNames: []string{"localhost"},
+	}
+
+	caPrivateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return nil, err
+	}
+
+	return &CertificateCreator{
+		ca:           ca,
+		caPrivateKey: caPrivateKey,
+	}, nil
+}
+
+func (cc *CertificateCreator) getEncodedCA() ([]byte, error) {
```

Comment:
by the name I thought this encodes the signer CA while it actually generates a new self-signed cert - could use a better name but maybe it could just live in the constructor?